app/models/profile.rb

```diff
@@ -147,26 +147,6 @@ def deactivate_for_gpo_verification
   end
 
   def deactivate_for_fraud_review
-    ##
-    # This is temporary. We are working on changing the way fraud review status
-    # is computed. The goal is that a profile is only in fraud review when
-    # `fraud_review_pending_at` is set. We will set this immediatly if a user
-    # verifies with phone and when a user enters their GPO code.
-    #
-    # We currently look at `fraud_pending_reason` to determine if a user is in
-    # fraud review. This allows us to change the writes on
-    # `fraud_review_pending_at` without side-effects.
-    #
-    # Once the writes on `fraud_review_pending_at` are correct we can move the
-    # reads to determine a user is fraud review pending to that column. At that
-    # point we can set `fraud_pending_reason` when we create a profile and
-    # deactivate the profile at the appropriate time for the given context
-    # (i.e. immediatly for phone and after GPO code entry for GPO).
-    #
-    if fraud_pending_reason.nil?
-      raise 'Attempting to deactivate a profile with a nil fraud pending reason'
-    end
```

Comment:
Is the idea that `fraud_pending_reason` should now be cleared before activation? I could see a case for still requiring it here and clearing it in the update so it's an atomic transaction.

Reply:
Fraud pending reason will be cleared before activation yes. This codes should not be relevant there though because this is deactivation.

I added this as a safeguard to make sure that profiles where not being deactivated without a fraud pending reason while I was moving things down. I'm cleaning it up now that I'm done moving things around and feel confident that we are updating the fraud pending reason and timestamps correctly.